iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/digitaltwin/DigitalTwinClientComponentTests.java

```diff
@@ -0,0 +1,211 @@
+    private static RegistryManager registryManager;
+    private String deviceId;
+    private DeviceClient deviceClient;
+    private DigitalTwinClient digitalTwinClient = null;
+    private static final String DEVICE_ID_PREFIX = "DigitalTwinServiceClientTests_";
+
+    @Rule
+    public Timeout globalTimeout = Timeout.seconds(5 * 60); // 5 minutes max per method tested
+
+    @Parameterized.Parameter(0)
+    public IotHubClientProtocol protocol;
+
+    @Parameterized.Parameters(name = "{index}: Digital Twin Test: protocol={0}")
+    public static Collection<Object[]> data() {
+        return asList(new Object[][] {
+                {MQTT},
+                {MQTT_WS},
+        });
+    }
+
+    @BeforeClass
+    public static void setUpBeforeClass() throws IOException {
+        registryManager = RegistryManager.createFromConnectionString(IOTHUB_CONNECTION_STRING);
+    }
+
+    @Before
+    public void setUp() throws URISyntaxException, IOException, IotHubException {
+
+        this.deviceClient = createDeviceClient(protocol);
+        deviceClient.open();
+        digitalTwinClient = DigitalTwinClient.createFromConnectionString(IOTHUB_CONNECTION_STRING);
+    }
+
+    @After
+    public void cleanUp() {
+        try {
+            deviceClient.closeNow();
+            registryManager.removeDevice(deviceId);
+        } catch (Exception ex) {
+            log.error("An exception occurred while closing/ deleting the device {}: {}", deviceId, ex);
+        }
+    }
+
+    private DeviceClient createDeviceClient(IotHubClientProtocol protocol) throws IOException, IotHubException, URISyntaxException {
+        ClientOptions options = new ClientOptions();
+        options.setModelId(E2ETestConstants.TEMPERATURE_CONTROLLER_MODEL_ID);
+
+        this.deviceId = DEVICE_ID_PREFIX.concat(UUID.randomUUID().toString());
+        Device device = Device.createDevice(deviceId, AuthenticationType.SAS);
+        Device registeredDevice = registryManager.addDevice(device);
+        String deviceConnectionString = registryManager.getDeviceConnectionString(registeredDevice);
+        return new DeviceClient(deviceConnectionString, protocol, options);
+    }
+
+    @AfterClass
+    public static void cleanUpAfterClass()
+    {
+        registryManager.close();
+    }
+
+    @Test
+    public void getDigitalTwin() {
+        // act
+        BasicDigitalTwin response = this.digitalTwinClient.getDigitalTwin(deviceId, BasicDigitalTwin.class);
+        ServiceResponseWithHeaders<BasicDigitalTwin, DigitalTwinGetHeaders> responseWithHeaders = this.digitalTwinClient.getDigitalTwinWithResponse(deviceId, BasicDigitalTwin.class);
+
+        // assert
+        assertEquals(response.getMetadata().getModelId(), E2ETestConstants.TEMPERATURE_CONTROLLER_MODEL_ID);
+        assertEquals(responseWithHeaders.body().getMetadata().getModelId(), E2ETestConstants.TEMPERATURE_CONTROLLER_MODEL_ID);
+    }
+
+    @Test
+    public void updateDigitalTwin() throws IOException {
+    }
```

Comment:
empty test

Reply:
yes it will be filled up in the next PR. I got tired and taking  a break before finishing update E2E test.